src/no-status-code-assert.ts

```diff
@@ -0,0 +1,103 @@
+// no-status-code-assert.ts
+
+/*
+ * Copyright (c) 2022-2024 Check Digit, LLC
+ *
+ * This code is licensed under the MIT license (see LICENSE.txt for details).
+ */
+
+import { StatusCodes } from 'http-status-codes';
+
+import { AST_NODE_TYPES, ESLintUtils, TSESLint, TSESTree } from '@typescript-eslint/utils';
+import getDocumentationUrl from './get-documentation-url';
+
+export const ruleId = 'no-status-code-assert';
+const NO_STATUS_CODE_ASSERT = 'NO_STATUS_CODE_ASSERT';
+
+const statusCodes = [
+  StatusCodes.OK,
+  StatusCodes.CREATED,
+  StatusCodes.BAD_REQUEST,
+  StatusCodes.UNAUTHORIZED,
+  StatusCodes.FORBIDDEN,
+  StatusCodes.NOT_FOUND,
+  StatusCodes.CONFLICT,
+  StatusCodes.INTERNAL_SERVER_ERROR,
+];
+
+const createRule = ESLintUtils.RuleCreator((name) => getDocumentationUrl(name));
+
+// Checks if a given AST node contains any status code value.
+const hasStatusCodeOrValue = (arg: TSESTree.Node): boolean => {
+  const isStatusCodeLiteral = (value: number): boolean => statusCodes.includes(value);
+
+  switch (arg.type) {
+    case AST_NODE_TYPES.Literal:
+      if (typeof arg.value === 'number') {
+        return isStatusCodeLiteral(arg.value);
```

Comment:
this branch doesn't have test coverage, either add test case or remove this logic and leave it for a separated rule to detect the direct usage of numeric value instead of http-status-codes for response status code?

Reply:
oh ya we are just focusing on statusCodes, we can remove this